src/circshiftedarray.jl

```diff
@@ -45,44 +53,29 @@ CircShiftedVector(v::AbstractVector, n = (0,)) = CircShiftedArray(v, n)
 
 size(s::CircShiftedArray) = size(parent(s))
 
-@inline function bringwithin(idx::Int, range::AbstractRange)
-    a, b = extrema(range)
-    n = length(range)
-    while idx < a
-        idx += n
-    end
-    while idx > b
-        idx -= n
+
+@inline function bringwithin(ind_with_offset::Int, ranges::AbstractUnitRange)
+    if ind_with_offset < first(ranges)
+        return ind_with_offset .- first(ranges) .+ 1 .+ last(ranges)
+    else
+        return ind_with_offset
```

Comment:
```suggestion
    return ifelse(ind_with_offset < first(ranges), ind_with_offset + length(ranges), ind_with_offset)
```
(no need for `broadcast` or branching). Here it should actually be `range`, because it is only one.